cmd/mcp/main.go

```diff
@@ -0,0 +1,183 @@
+		"The port to use for the MCP server. This is only used when the transport is set to 'sse'.")
+	addKubeConfigFlags(rootCmd)
+	rootCmd.SetOut(os.Stdout)
+	rootCmd.AddCommand(serveCmd)
+}
+
+func main() {
+	log.SetFlags(0)
+	ctrllog.SetLogger(logr.New(ctrllog.NullLogSink{}))
+
+	if err := rootCmd.Execute(); err != nil {
+		rootCmd.PrintErrf("✗ %v\n", err)
+		os.Exit(1)
+	}
+}
+
+// addKubeConfigFlags maps the kubectl config flags to the given persistent flags.
+// The default namespace is set to the value found in current kubeconfig context.
+func addKubeConfigFlags(cmd *cobra.Command) {
+	namespace := "default"
+	// Try to read the default namespace from the current context.
+	if ns, _, err := kubeconfigArgs.ToRawKubeConfigLoader().Namespace(); err == nil {
+		namespace = ns
+	}
+	kubeconfigArgs.Namespace = &namespace
+
+	cmd.PersistentFlags().StringVar(kubeconfigArgs.KubeConfig, "kubeconfig", getCurrentKubeconfigPath(),
+		"Path to the kubeconfig file.")
+	cmd.PersistentFlags().StringVar(kubeconfigArgs.Context, "kube-context", "",
+		"The name of the kubeconfig context to use.")
+	cmd.PersistentFlags().StringVar(kubeconfigArgs.Impersonate, "kube-as", "",
+		"Username to impersonate for the operation. User could be a regular user or a service account in a namespace.")
+	cmd.PersistentFlags().StringArrayVar(kubeconfigArgs.ImpersonateGroup, "kube-as-group", nil,
+		"Group to impersonate for the operation, this flag can be repeated to specify multiple groups.")
+	cmd.PersistentFlags().StringVar(kubeconfigArgs.ImpersonateUID, "kube-as-uid", "",
+		"UID to impersonate for the operation.")
+	cmd.PersistentFlags().StringVar(kubeconfigArgs.BearerToken, "kube-token", "",
+		"Bearer token for authentication to the API server.")
+	cmd.PersistentFlags().StringVar(kubeconfigArgs.APIServer, "kube-server", "",
+		"The address and port of the Kubernetes API server.")
+	cmd.PersistentFlags().StringVar(kubeconfigArgs.TLSServerName, "kube-tls-server-name", "",
+		"Server name to use for server certificate validation. If it is not provided, the hostname used to contact the server is used.")
+	cmd.PersistentFlags().StringVar(kubeconfigArgs.CertFile, "kube-client-certificate", "",
+		"Path to a client certificate file for TLS.")
+	cmd.PersistentFlags().StringVar(kubeconfigArgs.KeyFile, "kube-client-key", "",
+		"Path to a client key file for TLS.")
+	cmd.PersistentFlags().StringVar(kubeconfigArgs.CAFile, "kube-certificate-authority", "",
+		"Path to a cert file for the certificate authority.")
+	cmd.PersistentFlags().BoolVar(kubeconfigArgs.Insecure, "kube-insecure-skip-tls-verify", false,
+		"if true, the Kubernetes API server's certificate will not be checked for validity. This will make your HTTPS connections insecure.")
+	cmd.PersistentFlags().StringVarP(kubeconfigArgs.Namespace, "namespace", "n",
+		*kubeconfigArgs.Namespace, "The the namespace scope for the operation.")
+}
+
+func getCurrentKubeconfigPath() string {
+	defaultPath := ""
```

Comment:
This could be `filepath.Join(homedir.HomeDir(), ".kube", "config")`, where `filepath` is `path/filepath` and `homedir` is `k8s.io/client-go/util/homedir`

Reply:
This is copy/pasted from the CLI but here we should only look at the env var.